src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+/**
+ * Decorates the underlying filesystem by wrapping all calls to it with tracing.
+ *
+ * Parameters such as paths, directories or options are attached to the span as data,
+ * parameters that contain file contents are omitted due to potential problems with
+ * payload size or sensitive data.
+ */
+class TracingFilesystem implements Filesystem
+{
+    /** @var Filesystem */
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @param list<mixed> $args
+     * @param array<string, mixed> $data
+     */
+    protected function withTracing(string $method, array $args, array $data)
+    {
+        $context = new SpanContext();
+        $context->setOp("file.{$method}"); // See https://develop.sentry.dev/sdk/performance/span-operations/#web-server
+        $context->setDescription(json_encode($data, JSON_PRETTY_PRINT));
+        $context->setData($data);
+
+        return trace(function () use ($method, $args) {
+            return $this->filesystem->{$method}(...$args);
+        }, $context);
+    }
+
+    /** @see \Illuminate\Filesystem\FilesystemAdapter::assertExists() */
+    public function assertExists($path, $content = null)
```

Comment:
Why would we want to add these if we already have the `__call` these are test only methods right don't see much point in tracing them?

Reply:
> these are test only methods right

I think that is their intended purpose, but that does not mean that everybody is going to use them that way. Given they are present in `Illuminate\Support\Facades\Storage`, I could see an uninformed developer using them in live code.

Reply:
Hmmmmmmm. Okay sure, they won't hurt I guess.